standalone/regex.c

```diff
@@ -1,162 +1,237 @@
+    {
+    ovector = (int *)malloc(sizeof(int) * nmatch * 3);
+    if (ovector == NULL) return AP_REG_ESPACE;
+    allocated_ovector = 1;
+    }
+  }
+
+#ifdef WITH_PCRE2
+{
+  PCRE2_SPTR pcre2_s;
+  int pcre2_ret;
+  pcre2_match_data *match_data;
+  PCRE2_SIZE *pcre2_ovector = NULL;
+
+  pcre2_s = (PCRE2_SPTR)string;
+  match_data = pcre2_match_data_create_from_pattern(preg->re_pcre, NULL);
+  pcre2_match_context *match_context = pcre2_match_context_create(NULL);
+
+  pcre2_ret = pcre2_match((const pcre2_code *)preg->re_pcre, pcre2_s, (int)strlen(string),
+      0, (uint32_t)options, match_data, match_context);
+
+  if (match_data != NULL) {
+      pcre2_ovector = pcre2_get_ovector_pointer(match_data);
+      if (pcre2_ovector != NULL) {
+          for (int i = 0; ((i < pcre2_ret) && ((i*2) <= nmatch * 3)); i++) {
+              if ((i*2) < nmatch * 3) {
+                  ovector[2*i] = pcre2_ovector[2*i];
+                  ovector[2*i+1] = pcre2_ovector[2*i+1];
+              }
+          }
+      }
+      pcre2_match_data_free(match_data);
+      pcre2_match_context_free(match_context);
+  }
+}
+#else
+rc = pcre_exec((const pcre *)preg->re_pcre, NULL, string, (int)strlen(string),
+  0, options, ovector, nmatch * 3);
+#endif
+
+if (rc == 0) rc = nmatch;    /* All captured slots were filled in */
+
+if (rc >= 0)
+  {
+  apr_size_t i;
+  for (i = 0; i < (apr_size_t)rc; i++)
+    {
+    pmatch[i].rm_so = ovector[i*2];
+    pmatch[i].rm_eo = ovector[i*2+1];
+    }
+  if (allocated_ovector) free(ovector);
+  for (; i < nmatch; i++) pmatch[i].rm_so = pmatch[i].rm_eo = -1;
+  return 0;
+  }
+
+else
+  {
+  if (allocated_ovector) free(ovector);
+  switch(rc)
+    {
+#ifdef WITH_PCRE2
+    case PCRE2_ERROR_NOMATCH: return AP_REG_NOMATCH;
+    case PCRE2_ERROR_NULL: return AP_REG_INVARG;
+    case PCRE2_ERROR_BADOPTION: return AP_REG_INVARG;
+    case PCRE2_ERROR_BADMAGIC: return AP_REG_INVARG;
+    // case PCRE2_ERROR_UNKNOWN_NODE: return AP_REG_ASSERT; not defined in PCRE2
+    case PCRE2_ERROR_NOMEMORY: return AP_REG_ESPACE;
+#ifdef PCRE2_ERROR_MATCHLIMIT
+    case PCRE2_ERROR_MATCHLIMIT: return AP_REG_ESPACE;
+#endif
+#ifdef PCRE2_ERROR_BADUTF8
+    case PCRE2_ERROR_BADUTF8: return AP_REG_INVARG;
+#endif
+#ifdef PCRE2_ERROR_BADUTF8_OFFSET
+    case PCRE2_ERROR_BADUTF8_OFFSET: return AP_REG_INVARG;
+#endif
+#else // with old PCRE
+    case PCRE_ERROR_NOMATCH: return AP_REG_NOMATCH;
+    case PCRE_ERROR_NULL: return AP_REG_INVARG;
+    case PCRE_ERROR_BADOPTION: return AP_REG_INVARG;
+    case PCRE_ERROR_BADMAGIC: return AP_REG_INVARG;
+    case PCRE_ERROR_UNKNOWN_NODE: return AP_REG_ASSERT;
+    case PCRE_ERROR_NOMEMORY: return AP_REG_ESPACE;
+#ifdef PCRE_ERROR_MATCHLIMIT
+    case PCRE_ERROR_MATCHLIMIT: return AP_REG_ESPACE;
+#endif
+#ifdef PCRE_ERROR_BADUTF8
+    case PCRE_ERROR_BADUTF8: return AP_REG_INVARG;
+#endif
+#ifdef PCRE_ERROR_BADUTF8_OFFSET
+    case PCRE_ERROR_BADUTF8_OFFSET: return AP_REG_INVARG;
+#endif
+#endif // end of WITH_PCRE
```

Comment:
```suggestion
#endif
```

Reply:
Same as [here](https://github.com/owasp-modsecurity/ModSecurity/pull/3377#discussion_r2083110071).

Reply:
Line 221 already has the comment.